CodeReview/src/CodeReviewManager.cpp

```diff
@@ -63,4 +69,42 @@ QList<QList<VersionControlUI::DiffFrame*>> CodeReviewManager::orderDiffFrames(
 	return result;
 }
 
+void CodeReviewManager::saveReview(QString newVersion)
+{
+	auto store = new FilePersistence::SimpleTextFileStore{"."};
+	auto list = new Model::List{};
+	auto manager = new Model::TreeManager{CODE_REVIEW_COMMENTS_PREFIX+newVersion, list};
+
+	manager->beginModification(list, "set");
+	for (auto nodeID :  commentedNodes_.keys())
+	{
+		auto comment = commentedNodes_[nodeID];
+		// if a parent was set, it was a list from an earlier save
+		comment->setParent(nullptr);
```

Comment:
hmm, you can't simply set the parent to nullptr. If this works right now, chances are it does because the parent was already nullptr. The correct way to do this is to remove the element from the previous list (which you assume is the parent).